easybuild/framework/easyblock.py

```diff
@@ -358,34 +359,49 @@ def get_checksum_for(self, checksums, filename=None, index=None):
         :param filename: name of the file to obtain checksum for
         :param index: index of file in list
         """
-        checksum = None
-
-        # sometimes, filename are specified as a dict
+        chksum_input = filename
+        chksum_input_git = None
+        # if filename is provided as dict, take 'filename' key
         if isinstance(filename, dict):
-            filename = filename['filename']
+            chksum_input = filename['filename']
+            chksum_input_git = filename.get('git_config', None)
+        # early return if no filename given
+        if chksum_input is None:
+            self.log.debug("Cannot get checksum without a file name")
+            return None
+
+        if sys.version_info[0] >= 3 and sys.version_info[1] < 9:
+            # ignore any checksum for given filename due to changes in python/cpython#90021
+            # checksums of tarballs made by EB of git repos cannot be reliably checked prior to Python 3.9
+            if chksum_input_git is not None:
+                self.log.deprecated(
+                    "Reproducible tarballs of git repos are only supported in Python 3.9+. "
```

Comment:
```suggestion
                    "Reproducible tarballs of Git repos are only possible when using Python 3.9+ to run EasyBuild. "

Reply:
fixed in https://github.com/easybuilders/easybuild-framework/pull/4660/commits/4f30b16e2a60d91e1d35c87027270e993bc4d74b